api/models/assetscanstatus.go

```diff
@@ -0,0 +1,119 @@
+package models
+
+import (
+	"fmt"
+	"time"
+)
+
+var assetScanStatusStateTransitions = map[AssetScanStatusState][]AssetScanStatusState{
+	AssetScanStatusStatePending: {
+		AssetScanStatusStateScheduled,
+	},
+	AssetScanStatusStateScheduled: {
+		AssetScanStatusStateReadyToScan,
+		AssetScanStatusStateFailed,
+	},
+	AssetScanStatusStateReadyToScan: {
+		AssetScanStatusStateInProgress,
+		AssetScanStatusStateFailed,
+		AssetScanStatusStateAborted,
+	},
+	AssetScanStatusStateInProgress: {
+		AssetScanStatusStateDone,
+		AssetScanStatusStateFailed,
+		AssetScanStatusStateAborted,
+	},
+	AssetScanStatusStateAborted: {
+		AssetScanStatusStateFailed,
+	},
+}
+
+var assetScanStatusReasonMapping = map[AssetScanStatusState][]AssetScanStatusReason{
+	AssetScanStatusStatePending: {
+		AssetScanStatusReasonCreated,
+	},
+	AssetScanStatusStateScheduled: {
+		AssetScanStatusReasonScheduled,
+	},
+	AssetScanStatusStateReadyToScan: {
+		AssetScanStatusReasonOutOfVMClarity,
+		AssetScanStatusReasonReadyToScan,
+	},
+	AssetScanStatusStateInProgress: {
+		AssetScanStatusReasonInProgress,
+	},
+	AssetScanStatusStateAborted: {
+		AssetScanStatusReasonAborted,
+	},
+	AssetScanStatusStateFailed: {
+		AssetScanStatusReasonFailed,
+		AssetScanStatusReasonAbortTimedOut,
+	},
+	AssetScanStatusStateDone: {
+		AssetScanStatusReasonDone,
+	},
+}
+
+func NewAssetScanStatus(s AssetScanStatusState, r AssetScanStatusReason, m *string) *AssetScanStatus {
+	return &AssetScanStatus{
+		State:              s,
+		Reason:             r,
+		Message:            m,
+		LastTransitionTime: time.Now(),
+	}
+}
+
+func (as *AssetScanStatus) Equals(a *AssetScanStatus) bool {
+	if as.Message == nil && a.Message != nil {
+		return false
+	}
+	if a.Message == nil && as.Message != nil {
```

Comment:
nit: reusing the condition order makes it clearer
```suggestion
	if as.Message != nil && a.Message == nil {
```